tools/checkpoint_conversion/convert_gpt_neox_checkpoints.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import os
+
+import requests
+from transformers import GPTNeoXModel
+
+from keras_nlp.models.gpt_neox.gpt_neox_backbone import GPTNeoXBackbone
+
+PRESET_NAME = "pythia-70m"
+PRESET = "EleutherAI/pythia-70m-deduped"
+EXTRACT_DIR = "./{}"
+
+
+extract_dir = EXTRACT_DIR.format(PRESET_NAME)
+if not os.path.exists(extract_dir):
+    os.makedirs(extract_dir)
+
+# Config.
+config_path = os.path.join(extract_dir, "config.json")
+response = requests.get(f"https://huggingface.co/{PRESET}/raw/main/config.json")
+open(config_path, "wb").write(response.content)
+
+# Vocab.
+spm_path = os.path.join(extract_dir, "spm.model")
+response = requests.get(
+    f"https://huggingface.co/{PRESET}/resolve/main/spm.model"
+)
+open(spm_path, "wb").write(response.content)
+
+cfg = {}
+with open(config_path, "r") as pt_cfg_handler:
+    pt_cfg = json.load(pt_cfg_handler)
+
+cfg["vocabulary_size"] = pt_cfg["vocab_size"]
+cfg["num_layers"] = pt_cfg["num_hidden_layers"]
+cfg["num_heads"] = pt_cfg["num_attention_heads"]
+cfg["hidden_dim"] = pt_cfg["hidden_size"]
+cfg["intermediate_dim"] = pt_cfg["intermediate_size"]
+cfg["max_sequence_length"] = pt_cfg["max_position_embeddings"]
+cfg["layer_norm_epsilon"] = pt_cfg["layer_norm_eps"]
+cfg["rotary_pct"] = pt_cfg["rotary_pct"]
+cfg["rotary_emb_dim"] = pt_cfg["rotary_emb_dim"]
+
+hf_model = GPTNeoXModel.from_pretrained(PRESET)
```

Comment:
eventually we will probably want to extend this script to run some inputs through and check outputs, similar to other script in this directory

Reply:
Yes, I'll add that !